app/vmselect/netstorage/netstorage.go

```diff
@@ -1726,6 +1729,18 @@ func (snr *storageNodesRequest) collectResults(partialResultsCounter *metrics.Co
 		result := <-snr.resultsCh
 		if err := f(result.data); err != nil {
 			snr.finishQueryTracer(result.qt, fmt.Sprintf("error: %s", err))
+			resultsCollected++
+			if *skipSlowReplicas && resultsCollected > len(sns)-*replicationFactor {
```

Comment:
Why this code block has been moved from line 1756 to line 1733? It looks like it breaks the expected logic when `-search.skipSlowReplicas` command-line flag is set. In this case it should return response without waiting for the slowest vmstorage nodes if more that `len(sns)-*replicationFactor` vmstorage nodes successfully return response. In reality it returns the response only after some of `vmstorage` nodes return error.

Returned back the code block to the previous place in e1a2404db5e8e1248099750c6a4e45e90e2fa7be .

Reply:
That was a copy&paste mistake and supposed to be addressed here https://github.com/VictoriaMetrics/VictoriaMetrics/pull/4601#pullrequestreview-1520828732.

Thanks for pointing it out.